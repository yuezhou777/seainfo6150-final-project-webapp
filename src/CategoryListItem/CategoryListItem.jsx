import React from 'react';
import {Link} from "react-router-dom";
import HTMLText from "../HTMLText/HTMLText";
import styles from './CategoryListItem.module.css';

const CategoryListItem = (props) => {

    const category = props.breed;
    const posts = props.posts

    return(

        <div>

            <div className={styles.listContainer}>
                <h1>{category}</h1>

                <ul>
                    {posts
                        .filter(post => post.breed === category)
                        .map(post => (
                            <div className={styles.postContainer}>
                                {/*<h2><a href={'/seainfo6150-final-project-webapp/posts/:'+post.id}>{post.title}</a></h2>*/}
                                {/*<h2 className={styles.postTitle}><a href={'/seainfo6150-final-project-webapp/posts/'+post.id}>{post.title}</a></h2>*/}
                                <h2 className={styles.postTitle}>
                                    <Link to={'/posts/'+post.id}>{post.title}</Link>
                                </h2>

                                <p>{post.age} - {post.sex}</p>

                                <div>
                                <HTMLText content={post.intro} />
                                </div>
                            </div>
                        ))

                    }
                </ul>
            </div>

        </div>

    );

}

export default CategoryListItem;