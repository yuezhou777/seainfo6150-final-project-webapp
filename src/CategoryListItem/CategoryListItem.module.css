.listContainer {
    margin: 50px 300px 0px 300px;
}

.postContainer {
    margin: auto;
    padding: 20px 0px 30px 0px;
}