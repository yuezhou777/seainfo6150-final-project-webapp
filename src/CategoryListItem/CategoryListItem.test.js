import React from 'react'
import {render} from "@testing-library/react";
import CategoryListItem from "./CategoryListItem.jsx";
import { BrowserRouter } from 'react-router-dom';

describe('CategoryListItem test', () => {

    it('render CategoryListItem successfully ', () => {

        const category = 'Husky';
        const posts = [
            {
                "id":"1",
                "title":"Mel",
                "intro":"The only thing better than one adorable husky is an entire sled team of 'em. ",
                "content":"Mel is a gentle and sweet dog. She takes food out of your hand gently and likes to give little kisses, she still likes to run and play as if she was a puppy. Once she gets to know someone, she becomes less shy and more affectionate. She enjoys staying by a person's side.",
                "age":"1 year",
                "sex":"Female",
                "image":{
                   "_url":"https://images.unsplash.com/photo-1605559141066-1549783d18e9?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=400&q=250",
                   "size":123
                },
                "breed":"Husky",
                "category":"dog"
            }
        ];

        const {container} = render(<BrowserRouter><CategoryListItem category={category} posts={posts} /></BrowserRouter>);

        expect(container).toMatchSnapshot();
    });

})