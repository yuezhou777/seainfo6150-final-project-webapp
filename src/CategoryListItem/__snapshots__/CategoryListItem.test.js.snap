// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryListItem test render CategoryListItem successfully  1`] = `
<div>
  <div>
    <div
      class="listContainer"
    >
      <h1 />
      <ul />
    </div>
  </div>
</div>
`;
